### PR TITLE
Map old error code to new error

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement
related #27490

### CODE REVIEW

File: pkg/util/merr/utils.go [hunk 1]
```diff
@@ -185,6 +188,9 @@ func OldCodeToMerr(code commonpb.ErrorCode) error {
 	case commonpb.ErrorCode_SegmentNotFound:
 		return ErrSegmentNotFound
 
+	case commonpb.ErrorCode_MetaFailed:
```
Comment (congqixia) [inline pkg/util/merr/utils.go:191]: why could we map this code?

Comment (yah01) [reply pkg/util/merr/utils.go:191]: Only flush manager checks this error, and it's actually channel not found

Comment (congqixia) [reply pkg/util/merr/utils.go:191]: But it make no sense to convert here without any comment, it looks like a "error conversion"

**Pull request merged**